website/content/specification/syntax/constraints.md

```diff
@@ -163,6 +163,21 @@ And the following document.
 
 The expect constraint would pass for each `sibling` in the `parent` named "p1", and would fail for each `sibling` in the `parent` named "p2".
 
+## `expect` constraints
+
+The `expect` constraint is a type of Metaschema constraint that restricts field or flag value(s) based on the evaluation of a test Metapath expression.
+
+The `@target` attribute of an `<expect>` constraint specifies the node(s) in a document instance whose value is restricted by the constraint.
+
+The `@test` attribute of an `<expect>` constraint specifies the logical condition to be evaluated against each value node resulting from evaluating the `@target`. This expression must evaluate to [a Metaschema boolean value](/specification/datatypes#boolean) `true` or `false`.
+
+When the `@test` expression evaluates to `true` for a value node, then the value node is considered valid and passes the constraint.
+
+When the `@test` expression evaluates to `false` for a value node, then the value node is considered not valid and fails the constraint.
+
+A constraint may have an optional [`@level`](#level) attribute and/or an optional child [`<message>`](#message) element to indicate severity and documentation explaining how the target nodes are invalid.
+
+If defined, the `<message>` value MUST be a [Metaschema string value](/specification/datatypes#string). It may contain a Metapath expression templates that start with `{`, contain a Metapath expression, and end with `}`.  When evaluating a template Metapath expression, the context of the Metapath [evaluation focus](#constraint-processing) will be the failing value node.
```

Comment:
```suggestion
If defined, the `<message>` value MUST be a [Metaschema string value](/specification/datatypes#string). It MAY contain a Metapath expression templates that starts with `{`, contains a Metapath expression, and ends with `}`.  When evaluating a template Metapath expression, the context of the Metapath [evaluation focus](#constraint-processing) MUST be the failing value node.
```

Reply:
Good feedback, I should have been tighter with the MAY/MUST keywords.